doc/source/user_guide/mapdl.rst

```diff
@@ -66,60 +132,158 @@ method. Here is an example:
         mapdl.run("*VWRITE,LABEL(1),VALUE(1,1),VALUE(1,2),VALUE(1,3)")
         mapdl.run("(1X,A8,'   ',F10.1,'  ',F10.1,'   ',1F5.3)")
 
-Note that macros created within PyMAPDL (rather than loaded from
-a file) do not appear to run correctly. For example, here is the macro
-``DISP`` created using the ``*CREATE`` command within APDL:
+
+You can then view the final response of the non-interactive context with the
+:attr:`Mapdl.last_response <ansys.mapdl.core.Mapdl.last_response>` attribute.
+
+Using :meth:`Mapdl.non_interactive() <ansys.mapdl.core.Mapdl.non_interactive>`
+can also be useful to run commands on the server side without the interaction
+of Python. This can speed up things greatly, but you should be aware of how
+APDL works. An interesting discussion about speed comparison between PyMAPDL and APDL
+can be found in `here <pymapdl_discussion_speed_pymapdl_mapdl_>`_.
+
+It is recommended to use
+:meth:`Mapdl.non_interactive() <ansys.mapdl.core.Mapdl.non_interactive>` with precaution.
+
+How non-interactive context manager works
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The :meth:`Mapdl.non_interactive() <ansys.mapdl.core.Mapdl.non_interactive>` is implemented
+as a `context manager <python_context_manager_>`_ which means that there are some actions
+happening when entering and exit the context.
+When entering the context, :class:`Mapdl <ansys.mapdl.core.mapdl._MapdlCore>` stops sending any APDL
+command to the MAPDL instance.
```

Comment:
```suggestion
When entering the context, the class:`Mapdl <ansys.mapdl.core.mapdl._MapdlCore>` instance stops sending any APDL
command to the MAPDL instance.
```
A noun should always follow a code entity. The noun describes the type of Python object for the code entity.